sparse/src/KokkosSparse_spgemm_handle.hpp

```diff
@@ -776,6 +776,48 @@ class SPGEMMHandle {
   }
 
   bool get_compression_step() { return is_compression_single_step; }
+
+ private:
+  // An SpGEMM handle can be reused for multiple products C = A*B, but only if
+  // the sparsity patterns of A and B do not change. Enforce this by recording
+  // the raw data pointers of the matrices' rowptrs and entries during symbolic
+  // and numeric, and make sure they never change.
+  const void *Arowptrs = nullptr, *Aentries = nullptr;
+  const void *Browptrs = nullptr, *Bentries = nullptr;
+  const void *Crowptrs = nullptr, *Centries = nullptr;
+
+ public:
+  bool checkMatrixIdentitiesSymbolic(const void *ArowptrsIn,
```

Comment:
Checking that pointers are the same is good but if the data that is pointed to is modified this will results in the sparsity of these matrices to potentially change?

Reply:
@brian-kelley , that sounds better to me. A debug-only check that doesn't rely on pointer comparisons.

Reply:
@lucbv @jgfouca I just added this check in debug builds.